test/extended/operators/routable.go

```diff
@@ -0,0 +1,129 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[Feature:Platform] Managed cluster should", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc = exutil.NewCLI("operators-routable", exutil.KubeConfigPath())
+
+		// routeHostWait is how long to wait for routes to be assigned a host
+		routeHostWait = 30 * time.Second
+
+		// endpointWait is how long to wait for endpoints to be reachable
+		endpointWait = 3 * time.Minute
+	)
+
+	g.BeforeEach(func() {
+		_, ns, err := exutil.GetRouterPodTemplate(oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "couldn't find default router")
+
+		svc, err := oc.AdminKubeClient().CoreV1().Services(ns).Get("router-default", metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				g.Skip("default router is not exposed by a load balancer service")
+			}
+			o.Expect(err).NotTo(o.HaveOccurred(), "error getting default router service: %v", err)
+		}
+
+		if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+			g.Skip("default router is not exposed by a load balancer service")
+		}
+	})
+
+	g.It("expose cluster services outside the cluster", func() {
+		ns := oc.KubeFramework().Namespace.Name
+
+		execPodName := exutil.CreateExecPodOrFail(oc.AdminKubeClient().Core(), ns, "execpod")
+		defer func() { oc.AdminKubeClient().Core().Pods(ns).Delete(execPodName, metav1.NewDeleteOptions(1)) }()
+
+		tests := []struct {
+			routeNamespace string
+			routeName      string
+			scheme         string
+			expect         string
+		}{
+			{
+				routeNamespace: "openshift-console",
+				routeName:      "console",
+				scheme:         "https",
+				expect:         "200",
+			},
+			{
+				routeNamespace: "openshift-monitoring",
+				routeName:      "prometheus-k8s",
+				scheme:         "https",
+				expect:         "403",
+			},
+		}
+		for _, test := range tests {
+			g.By(fmt.Sprintf("verifying the %s/%s route has an ingress host", test.routeNamespace, test.routeName))
+			var hostname string
+			err := wait.Poll(time.Second, routeHostWait, func() (bool, error) {
+				route, err := oc.AdminRouteClient().Route().Routes(test.routeNamespace).Get(test.routeName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if len(route.Status.Ingress) == 0 || len(route.Status.Ingress[0].Host) == 0 {
+					return false, nil
+				}
+				hostname = route.Status.Ingress[0].Host
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			url := fmt.Sprintf("%s://%s", test.scheme, hostname)
+			g.By(fmt.Sprintf("verifying the %s/%s route serves 200 from %s", test.routeNamespace, test.routeName, url))
+			err = waitForURLOK(ns, execPodName, url, test.expect, endpointWait)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+})
+
+func waitForURLOK(ns, execPodName, url, expect string, timeout time.Duration) error {
```

Comment:
This code is fine, but you might want to consider using the route tester harness

	"github.com/openshift/origin/test/extended/util/url"

Lets you do some more efficient testing:

```
			t.Within(
				3*time.Minute,
				url.Expect("GET", "http://1.ingress-test.com/test").Through(host).HasStatusCode(200),
				url.Expect("GET", "http://1.ingress-test.com/other/deep").Through(host).HasStatusCode(200),
				url.Expect("GET", "http://1.ingress-test.com/").Through(host).HasStatusCode(503),
				url.Expect("GET", "http://2.ingress-test.com/").Through(host).HasStatusCode(200),
				url.Expect("GET", "https://3.ingress-test.com/").Through(host).SkipTLSVerification().HasStatusCode(200),
				url.Expect("GET", "http://3.ingress-test.com/").Through(host).RedirectsTo("https://3.ingress-test.com/", http.StatusFound),
			)
```

Reply:
Great tip, thanks